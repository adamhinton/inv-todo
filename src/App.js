import Header from "./components/Header";
import { useState } from "react";
import styled from "styled-components";
import AddNewTaskForm from "./components/AddNewTaskForm";
import TodosContainer from "./components/TodosContainer";
import { useEffect } from "react";

function App() {
  const [todoList, setTodoList] = useState([]);

  useEffect(() => {
    const key = "todos";

    const persistedTodos = JSON.parse(window.localStorage.getItem(key)) || [];
    setTodoList(persistedTodos);
  }, []);

  return (
    <div className="App">
      <div className="container">
        <Header setTodoList={setTodoList} />
        <StyledMain>
          <AddNewTaskForm todoList={todoList} setTodoList={setTodoList} />
          <TodosContainer
            isCompleted={false}
            todoList={todoList}
            setTodoList={setTodoList}
          />
          <TodosContainer
            isCompleted={true}
            todoList={todoList}
            setTodoList={setTodoList}
          />
        </StyledMain>
      </div>
    </div>
  );
}

export default App;

const StyledMain = styled.main`
  width: 900px;

  @media (max-width: 1000px) {
    width: 70%;
  }

  @media (max-width: 850px) {
    width: 90%;
  }

  @media (max-width: 500px) {
    width: 100%;
  }
`;

export const persistTodosToLocalStorage = (todoList) => {
  const key = "todos";
  window.localStorage.setItem(key, JSON.stringify(todoList));
};
