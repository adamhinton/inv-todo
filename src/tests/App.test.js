import App from "../App";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";

test("[1] App renders without errors", () => {
  render(<App />);
});

test("[2] Matches snapshot from 5.29.22", () => {
  const component = renderer.create(<App />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
