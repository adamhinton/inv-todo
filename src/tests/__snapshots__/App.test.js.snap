// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[2] Matches snapshot from 5.29.22 1`] = `
<div
  className="App"
>
  <div
    className="container"
  >
    <header
      className="sc-bczRLJ crdXbW"
    >
      <h1
        className="sc-gsnTZi gWWeNP"
      >
        Inventora Tasks
      </h1>
      <button
        className="sc-dkzDqf bexeyA"
        onClick={[Function]}
      >
        Delete All
      </button>
    </header>
    <main
      className="sc-iqcoie hkmRiF"
    >
      <form
        className="sc-hKMtZM iYxyjv"
        onSubmit={[Function]}
      >
        <label
          className="sc-eCYdqJ kwJpGj"
        >
          Add New Task
        </label>
        <input
          className="sc-jSMfEi kCSPQW"
          data-testid="add-new-task-input"
          onChange={[Function]}
          placeholder="Enter Task Description..."
          type="text"
          value=""
        />
        <button
          className="sc-gKXOVf dpfsCR"
          type="submit"
        >
          + Add Task
        </button>
      </form>
      <section>
        <h2
          className="sc-kDDrLX kLBLqs"
        >
          To-Do
        </h2>
        <ul />
      </section>
      <section>
        <h2
          className="sc-kDDrLX kLBLqs"
        >
          Completed
        </h2>
        <ul />
      </section>
    </main>
  </div>
</div>
`;
